MonoDevelop.FSharpBinding/FSharpOutlineTextEditorExtension.fs

```diff
@@ -0,0 +1,127 @@
+                                       |> Array.sortBy(fun xs -> xs.Range.StartLine)
+
+                        for nested in children do
+                            treeStore.AppendValues(iter, [| nested |]) |> ignore
+
+                    treeView.ExpandAll())
+                Gdk.Threads.Leave()
+        | None -> ()
+
+        refreshingOutline <- false
+        false
+
+    interface IOutlinedDocument with
+        member x.GetOutlineWidget() =
+            match treeView with
+            | Some(treeView) -> treeView :> Widget
+            | None ->
+                let treeStore = new TreeStore(typedefof<obj>)
+                let padTreeView = new PadTreeView(treeStore)
+
+                let setCellIcon (_) (cellRenderer : CellRenderer) (treeModel : TreeModel) (iter : TreeIter) =
+                    let pixRenderer = cellRenderer :?> CellRendererImage
+                    let item = treeModel.GetValue(iter, 0) :?> FSharpNavigationDeclarationItem
+                    pixRenderer.Image <- ImageService.GetIcon(ServiceUtils.getIcon item.Glyph, Gtk.IconSize.Menu)
+
+                let setCellText (_) (cellRenderer : CellRenderer) (treeModel : TreeModel) (iter : TreeIter) =
+                    let renderer = cellRenderer :?> CellRendererText
+                    let item = treeModel.GetValue(iter, 0) :?> FSharpNavigationDeclarationItem
+                    renderer.Text <- item.Name
+                let jumpToDeclaration focus =
+                    let iter : TreeIter ref = ref Unchecked.defaultof<_>
+                    match padTreeView.Selection.GetSelected(iter) with
+                    | true -> let node = padTreeView.Model.GetValue(!iter, 0) :?> FSharpNavigationDeclarationItem
+                              let (scol,sline) = node.Range.StartColumn, node.Range.StartLine
+                              x.Editor.SetCaretLocation(max 1 sline, max 1 scol, true)
+                    | false -> ()
+                    if focus then
+                        x.Editor.GrabFocus()
+
+                treeView <- Some padTreeView
+
+                let pixRenderer = new CellRendererImage()
+                pixRenderer.Xpad <- 0u
+                pixRenderer.Ypad <- 0u
+                padTreeView.TextRenderer.Xpad <- 0u
+                padTreeView.TextRenderer.Ypad <- 0u
+
+                let treeCol = new TreeViewColumn()
+                treeCol.PackStart(pixRenderer, false)
+                treeCol.SetCellDataFunc(pixRenderer, new TreeCellDataFunc(setCellIcon))
+                treeCol.PackStart(padTreeView.TextRenderer, true)
+                treeCol.SetCellDataFunc(padTreeView.TextRenderer, new TreeCellDataFunc(setCellText))
+
+                padTreeView.AppendColumn treeCol |> ignore
+                padTreeView.HeadersVisible <- true
+                padTreeView.Realized.Add(fun _ -> x.refillTree |> ignore)
+                padTreeView.Selection.Changed.Subscribe(fun _ -> jumpToDeclaration false) |> ignore
```

Comment:
Maybe dont ignore the IDisposables but add them to a disposable collection and Dispose of them in IDisposable.Dipose method